src/pubsub/ua_pubsub_networkmessage_json.c

```diff
@@ -478,14 +479,14 @@ NetworkMessage_decodeJsonInternal(UA_NetworkMessage *dst, CtxJson *ctx,
     DecodeEntry entries[5] = {
         {UA_DECODEKEY_MESSAGEID, &dst->messageId, NULL, false, &UA_TYPES[UA_TYPES_STRING]},
         {UA_DECODEKEY_MESSAGETYPE, &messageType, NULL, false, NULL},
-        {UA_DECODEKEY_PUBLISHERID, &dst->publisherId.publisherIdString, NULL, false, pubIdType},
+        {UA_DECODEKEY_PUBLISHERID, &dst->publisherId.string, NULL, false, pubIdType},
         {UA_DECODEKEY_DATASETCLASSID, &dst->dataSetClassId, NULL, false, &UA_TYPES[UA_TYPES_GUID]},
         {UA_DECODEKEY_MESSAGES, &dst->payload.dataSetPayload.dataSetMessages, &DatasetMessage_Array_decodeJsonInternal, false, NULL}
     };
 
     //Store publisherId in correct union
     if(pubIdType == &UA_TYPES[UA_TYPES_UINT64])
-        entries[2].fieldPointer = &dst->publisherId.publisherIdUInt64;
+        entries[2].fieldPointer = &dst->publisherId.uint64;
```

Comment:
You don't need this at all.
Because string and uint64 have the same memory location in the union.
The set pubIdType is already correct.

Reply:
removed if clause